src/test/java/com/google/devtools/build/lib/skyframe/rewinding/RewindingTest.java

```diff
@@ -69,11 +80,24 @@ public void workspaceInit(
             });
   }
 
+  @Override
+  protected ImmutableList<BlazeModule> getSpawnModules() {
+    return ImmutableList.<BlazeModule>builder()
+        .addAll(super.getSpawnModules())
+        .add(new CredentialModule())
+        .build();
+  }
+
   @Override
   protected void setupOptions() throws Exception {
     super.setupOptions();
+    if (worker == null) {
+      worker = startWorker();
```

Comment:
`setupOptions()` is an odd place to start a worker. I assume you can't use a `@Before` method because you need it from a superclass `@Before`, so how about `runPriorToBeforeMethods()`?

Also, is it slow to start/stop a worker? If so, it could be a good idea to make it `static` and start/stop it once.

Seems like `WorkerInstance` could be an `ExternalResource` so that it can be used as an `@Rule` or `@ClassRule`, which would eliminate the timing concern.

Reply:
I refactored it into a `TestRule` that is used as both a `@Rule` and a `@ClassRule` in https://github.com/bazelbuild/bazel/pull/25432. The current PR is now stacked on it and the second commit only touches the rewinding test.

Reply:
Interesting, didn't know you could use both annotations together, but it is an intended use case: https://github.com/junit-team/junit/blob/master/doc/ReleaseNotes4.12.md#pull-request-932-allow-static-rules-also-annotated-with-classrule.